src/lib/permission-exclusions-store.ts

```diff
@@ -67,6 +77,23 @@ class PermissionExclusionResources {
     return this.permissionExclusionResourceData
   }
 
+  getConsolidatedTempData (resourceName: string, permissionsA: string[], permissionsB: string[]) : any {
+    // Create a deep clone of existing data
+    const tempPermissionExclusionResourceData = JSON.parse(JSON.stringify(this.permissionExclusionResourceData))
+    if (!tempPermissionExclusionResourceData[resourceName]) {
+      tempPermissionExclusionResourceData[resourceName] = {}
+    }
+    if (!tempPermissionExclusionResourceData[resourceName].permissionsA) {
+      tempPermissionExclusionResourceData[resourceName].permissionsA = []
+    }
+    if (!tempPermissionExclusionResourceData[resourceName].permissionsB) {
+      tempPermissionExclusionResourceData[resourceName].permissionsB = []
+    }
+    tempPermissionExclusionResourceData[resourceName].permissionsA = permissionsA
+    tempPermissionExclusionResourceData[resourceName].permissionsB = permissionsB
+    return tempPermissionExclusionResourceData
```

Comment:
Perhaps it's a matter of style, but I found this difficult to read. I'd probably have written it more like
```suggestion
    const tempPermissionExclusionResourceData = JSON.parse(JSON.stringify(this.permissionExclusionResourceData))
    return {
      ...tempPermissionExclusionResourceData,
      [resourceName]: {
        ...(tempPermissionExclusionResourceData[resourceName] || {}),
        permissionsA,
        permissionsB,
      },
    }
```
To me the shape of the result is clearer. I totally acknowledge you might disagree about readability of this though, so it's just a suggestion.